cmd/ingester/app/consumer/consumer.go

```diff
@@ -85,7 +85,8 @@ func (c *Consumer) Start() {
 				p.wg.Wait()
 			}
 			c.partitionIDToState[pc.Partition()] = &consumerState{partitionConsumer: pc}
-			c.partitionIDToState[pc.Partition()].wg.Add(2)
+			c.partitionIDToState[pc.Partition()].wg.Add(2) // we will be adding 2 goroutines
+			c.partitionsHeld.Update(int64(len(c.partitionIDToState)))
```

Comment:
@vprithvi this wouldn't quite work because we never seem to delete things from this map.